src/VisualStudio/Core/Def/ServicesVSResources.resx

```diff
@@ -977,4 +977,20 @@ Additional information: {1}</value>
   <data name="Search_found_no_results" xml:space="preserve">
     <value>Search found no results</value>
   </data>
+  <data name="Enable_navigation_to_decompiled_sources" xml:space="preserve">
+    <value>Enable navigation to decompiled sources (experimental)</value>
+  </data>
+  <data name="Decompiler_Legal_Notice_Message" xml:space="preserve">
+    <value>IMPORTANT:  VISUAL STUDIO INCLUDES DECOMPILING FUNCTIONALITY (“DECOMPILER”) THAT ENABLES REPRODUCING SOURCE CODE FROM BINARY CODE. BY ACCESSING AND USING THE DECOMPILER, YOU AGREE TO THE TERMS AND CONDITIONS IN THE VISUAL STUDIO LICENSE TERMS AND THE TERMS FOR THE DECOMPILER BELOW.  
```

Comment:
Without making a comment of the legalease for a moment (I am not a lawyer after all), should the first line of this message just be "you have enabled navigation to decompiled types, and we are about to navigate, are you sure?" followed with the full EULA? Imagine the customer that checked the box a week ago and only uses the feature the first time. The EULA is understandable once you have the context, but if you don't mentally connect this dialog to the checkbox you might be thrown off a bit.

I also admit we're telling them to verify they have license terms for the binary code without actually telling them which binary we're going to decompile. Quite the catch-22. :-)

Reply:
(I now see @sharwell's comment asking our legal team for a new text. So disregard this.)

Reply:
:arrow_right: Fixed in 1b96597c4104570b64588841914b6ac4338ac37e